### PR TITLE
community: SamabanovaCloud tool calling and Structured output

### PR DESCRIPTION
**Pull request by jhpiedrahitao**
**Description:** Add tool calling and structured output support for SambaNovaCloud chat models, docs included

### CODE REVIEW

**Comment by jhpiedrahitao** [comment]
@ccurme can you please take a look?

**Review by ccurme — APPROVED** [review]

File: docs/docs/integrations/chat/sambanova.ipynb [hunk 1]
```diff
@@ -120,10 +120,15 @@
    "metadata": {},
    "outputs": [],
    "source": [
-    "from langchain_community.chat_models.sambanova import ChatSambaNovaCloud\n",
+    "# from langchain_community.chat_models.sambanova import ChatSambaNovaCloud\n",
+    "from langchain_chat_models import ChatSambaNovaCloud\n",
```
Comment (ccurme) [inline docs/docs/integrations/chat/sambanova.ipynb:124]: This looks like a typo?

Comment (jhpiedrahitao) [reply docs/docs/integrations/chat/sambanova.ipynb]: Right thanks,  fixed

**Comment by jhpiedrahitao** [comment]
@ccurme typo was fixed

**Pull request merged**